test/libscap/test_suites/tables/syscall_table.cpp

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright (C) 2023 The Falco Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+#include <scap.h>
+#include <gtest/gtest.h>
+
+extern const struct syscall_evt_pair g_syscall_table[SYSCALL_TABLE_SIZE];
+
+/* Each syscall_id should have its own PPM_SC, note that this should be true also for generic syscalls
+ * only the event type is generic, the PPM_SC code is defined! This test is architecture dependent!
```

Comment:
Since we use ifdefs into the g_syscall_table this check should architecture dependent